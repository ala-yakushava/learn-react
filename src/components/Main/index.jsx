import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';
import Container from '../Container';
import MovieCtalog from '../MovieCatalog';
import Footer from '../Footer';

const Main = ({ children }) => (
  <div className="Main">
    <Container>
      {children}
    </Container>
    <div className="Main_space" />
    <Container>
      <MovieCtalog />
    </Container>
    <Footer />
  </div>
);

Main.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Main;
