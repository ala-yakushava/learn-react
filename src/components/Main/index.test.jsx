import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Main from './index';

jest.mock('../MovieCatalog', () => () => (
  <div data-testid="MovieCatalog" />
));

describe('Components - Main', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(
        <Main>
          <div>Content</div>
        </Main>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should be render MovieCatalog', () => {
    const { queryByTestId } = render(
      <Main>
        <div>Content</div>
      </Main>,
    );

    expect(queryByTestId('MovieCatalog')).toBeInTheDocument();
  });
});
