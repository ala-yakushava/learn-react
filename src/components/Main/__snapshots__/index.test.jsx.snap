// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Main should match snapshot 1`] = `
<div
  className="Main"
>
  <div
    className="Container"
  >
    <div>
      Content
    </div>
  </div>
  <div
    className="Main_space"
  />
  <div
    className="Container"
  >
    <div
      data-testid="MovieCatalog"
    />
  </div>
  <footer
    className="Footer"
  >
    <p
      className="Logo"
    >
      <b>
        netflix
      </b>
      roulette
    </p>
  </footer>
</div>
`;
